### ISSUE TITLE
ipv4 adress '192.168.0.2000000000' is legal

### ISSUE THREAD

**Issue by stdarg** [sindresorhus/ip-regex#18]
Was trying out your regular expressions. Did a little testing. It works great, except I'm pretty sure each octet is only 8 bits, and should not exceed a value of 255. That's from memory.

**Comment by Mottie** [comment]
Using the regular expression without an exact flag will report that an IP exists within the string. You'll need to include an `exact: true` while using the regular expression to determine if the address is valid:

```js
const ipRegex = require('ip-regex');

// Contains an IP address?
ipRegex().test('192.168.0.2000000000');
//=> true

// Is an IP address?
ipRegex({exact: true}).test('192.168.0.2000000000');
//=> false
```

It appears that #19 needs the same flag to provide the expected result.

**Comment by sindresorhus** [comment]
> Edit: Maybe we need to include boundaries (\b) in the regex when the exact parameter is false.

Yeah, I think that would be a good idea.

**Comment by Mottie** [comment]
Hmm, more complicated than I thought... adding `\b` isn't working because `:` isn't a word character, so the test fails on `::`, `1::` and `::1`.

**Comment by silverwind** [comment]
I think I have to oppose this change. The example

````js
// Contains an IP address?
ipRegex().test('192.168.0.2000000000');
//=> true
````
looks correct to me, it **does** contain a valid IP address and the correct solution to the user's problem is to use the `exact` option.

Also It seems this update broke certain cases of matching multiple addresses in a string:

````js
$ yarn add ip-regex@2 && node 
> "::1, ::2".match(require("ip-regex")())
[ '::1', '::2' ]
$ rm -rf node_modules/ip-regex
$ yarn add ip-regex@3 && node
> "::1, ::2".match(require("ip-regex")())
[ '::2' ]

**Comment by Mottie** [comment]
Or maybe add a `includeBoundaries` option?

**Comment by sindresorhus** [comment]
Fixed by #23
